projects/telemetry/scripts/interface.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+import serial
+import struct
+import ctypes
+
+class CANIdBits(ctypes.LittleEndianStructure):
+  _fields_ = [
+    ('source_id', ctypes.c_uint16, 4),
+    ('type', ctypes.c_uint16, 1),
+    ('msg_id', ctypes.c_uint16, 6)
+  ]
+
+class CANId(ctypes.Union):
+  _fields_ = [
+    ('b', CANIdBits),
+    ('as_u16', ctypes.c_uint16)
+  ]
+
+MSG_FORMAT = '<3cHQcc'
```

Comment:
Magic?

Reply:
It's the struct format code.
Little endian, 3 characters, unsigned 16-bit int, unsigned 64-bit int, 2 characters. :P
I can add a link to the format table in the python docs?

Reply:
That would be great :) Or even just a note that it is a format code for structs. I've never used/seen that before.